Encode.xs

```diff
@@ -139,6 +140,31 @@ do_fallback_cb(pTHX_ UV ch, SV *fallback_cb)
 }
 
 static SV *
+do_bytes_fallback_cb(pTHX_ U8 *s, STRLEN slen, SV *fallback_cb)
+{
+    dSP;
+    int argc;
+    STRLEN i;
+    SV *retval = newSVpvn("",0);
```

Comment:
It is not a good idea to make it mortal as mortal variables are freed later, when leaving encode/decode XS function. Function do_fallback_cb can be called too many times, each time for invalid character in input string. And input string is lot of times untrusted and can comes from attacker.

I tried it with mortal variables and following code
`perl -MEncode -e 'my $b = "\x{ff}" x 10000000; decode("UTF-8", $b, sub { "$_[0]" })'`
eats 825MB. Without mortal variables only 58MB. Which is a huge difference.

I looked at other XS code available in perl repository and it uses following pattern:
```
retval = POPs;
SvREFCNT_inc(retval);
...
FREETMPS;
```
Looking into perl source code, FREETMPS just remove tmp flag and decrements refcount of all variables placed in tmp stack and let stack empty. So SvREFCNT_inc prevents destroying retval in FREETMPS and make it as normal non-mortal variable. Looks like it could save some memory as new variable is not needed to allocate.

But reusing SV from POPs or creating new variable via newSVsv() is not enough. That Encode code expects that returned SV has valid PV slot. So this is reason why such construct `newSVpvn("",0); ... sv_catsv(retval, POPs);` is used.

I will try to cleanup this code and will use above pattern.